src/main/java/net/eldeen/dropwizard/CfSignalResourceBundle.java

```
@@ -7,75 +17,154 @@
+      }
+
+      return cloudFormationAtomicRef.updateAndGet((unused) -> {
+
+        AmazonCloudFormationClient amazonCloudFormationClient = new AmazonCloudFormationClient();
+
+        amazonCloudFormationClient.setRegion(Region.getRegion(Regions.valueOf(cfSignalResourceConfig.getAwsRegion()
+                                                                                                    .toUpperCase()
+                                                                                                    .replace('-', '_'))));
+        return amazonCloudFormationClient;
+      });
+    };
+  }
+
+  @Inject
+  public CfSignalResourceBundle(@CfSignalResourceInstanceId Optional<String> instanceId, AmazonCloudFormation amazonCloudFormation) {
+    instanceIdProvider = () -> checkNotNull(instanceId);
+    cloudFormationSupplier = (config) -> amazonCloudFormation;
   }
 
   @Override
   public void initialize(Bootstrap<?> bootstrap) {
+  }
 
+  public CfSignalResourceConfig getConfiguration() {
+    return new CfSignalResourceConfig();
   }
 
   @Override
-  public void run(Environment environment) {
+  public void run(T config, Environment environment) {
 
-    final String instanceId = EC2MetadataUtils.getInstanceId();
-    if (instanceId == null || instanceId.trim() == "") {
+    final Optional<String> instanceId = instanceIdProvider.get();
+    if (!instanceId.isPresent()) {
       LOGGER.debug("Unable to fetch EC2 Instance ID, assuming not running on AWS and thus not signalling");
       return;
     }
 
-    environment.lifecycle().addServerLifecycleListener(new ServerLifecycleListener() {
-      @Override
-      public void serverStarted(Server server) {
-        AmazonCloudFormation client = null;
+    environment.lifecycle()
+               .addLifeCycleListener(
+                 new CfSignalResourceLifcycleListener(getCfResourceBundleConfig(config).orElseGet(this::getConfiguration),
+                                                      instanceId.get()));
```

Not sure I like defaulting to an empty configuration. Would rather throw an exception or log a warning or something.

Either that or sanity check the values in the config object before making a request based on it.